packages/squiggle-lang/src/dists/SampleSetDist/index.ts

```diff
@@ -228,6 +228,32 @@ sample everything.
   static deserialize(value: readonly number[]): SampleSetDist {
     return new SampleSetDist(value);
   }
+
+  summarize(
+    keys: Array<
+      "mean" | "p50" | "p5" | "p25" | "p75" | "p95" | "stdev" | "min" | "max"
+    > = ["mean", "p5", "p50", "p95"]
```

Comment:
How general purpose this method is supposed to be?

Right now you only use it for simple values, but its interface is more flexible than that.

If the plan is to also use it for `<SummaryTable>`, then there's a problem with how we handle `result<...>` there compared to this code, and also the question of whether it should be on `SampleSetDist` or on the `BaseDist`.

If its main purpose if for simple values then I'd argue that it doesn't belong here and it's better to just call necessary methods from `simpleVaule` code.

My concern here is that when it's not clear what the contract is, we might end up with several slightly different methods for different tasks, and it'd be confusing.

I understand that the decision whether to `result<...>` or not to `result<...>` is annoying, though, especially because on sample sets specifically it's guaranteed to be ok. And also I see that the "sorted" abstraction is leaking (did you avoid going through `this.inv(...)` for performance reasons?)